src/coreclr/debug/shared/dbgtransportsession.cpp

```diff
@@ -700,15 +705,12 @@ HRESULT DbgTransportSession::SendMessage(Message *pMessage, bool fWaitsForReply)
                 pMessage->m_pNext = NULL;
             }
         }
-
-        // If the state is SS_Open we can send the message now.
-        if (m_eState == SS_Open)
+        else
```

Comment:
At this point, we're either adding the message to the queue, where the queue reader is responsible for freeing.

Or `else` we delete when we've allocated a copy.